internal/provider_test/integration_test.go

```diff
@@ -49,6 +51,19 @@ func (option TestOption) IsFalse() bool {
 	return option.Get() == "false"
 }
 
+func (option TestOption) GetSemver() (*semver.Version, error) {
+	return semver.NewVersion(option.Get())
+}
+
+func (option TestOption) IsVersionLessThan(version string) bool {
+	thisVersion, err := option.GetSemver()
+	if err != nil {
+		return false
+	}
+	suppliedVersion, err := semver.NewVersion(version)
+	return err == nil && thisVersion.LessThan(suppliedVersion)
```

Comment:
Why return a `false` if you cannot parse one of the version strings? I would think that that is a test defect that should not be ignored but should fail the test.

Reply:
The version can be the empty string (if the environment variable was not set) or something else that does not parse as semver for some reason. It is much easier to make sure that the versions are valid semver in the cases where I want to check the version to skip the test than in the case where I want to run the test (all of them exception 2.4.1 variants). If the return value of true causes a test to run that should have been skipped, then we should find out when the test fails.

Reply:
You mean that `option` could be empty, right? `version` should always parse correctly.